homeassistant/components/shelly/utils.py

```diff
@@ -73,16 +78,17 @@ def get_number_of_channels(device: BlockDevice, block: Block) -> int:
 def get_block_entity_name(
     device: BlockDevice,
     block: Block | None,
-    description: str | DeviceClassName | None = None,
+    description: str | DeviceClassName | DeviceName | None = None,
 ) -> str:
     """Naming for block based switch and sensors."""
     channel_name = get_block_channel_name(device, block)
 
     if description:
-        # It's not possible to do string manipulations on DEVICE_CLASS_NAME
-        # the assert satisfies the type checker and will catch attempts
-        # to use DEVICE_CLASS_NAME as description.
+        # It's not possible to do string manipulations on DEVICE_CLASS_NAME or
+        # DEVICE_NAME. The asserts satisfy the type checker and will catch attempts
+        # to use DEVICE_CLASS_NAME or DEVICE_NAME as descriptions.
         assert description is not DEVICE_CLASS_NAME
+        assert description is not DEVICE_NAME
```

Comment:
This looks like a potentially evergrowing list. Perhaps some type is not in () with the list of special names defined in const?

Reply:
It doesn't work, mypy complains about this:
```diff
-        assert description is not DEVICE_CLASS_NAME
-        assert description is not DEVICE_NAME
+       assert description not in (DEVICE_CLASS_NAME, DEVICE_NAME)
```
```
homeassistant/components/shelly/utils.py:91: error: Item "DeviceClassName" of "Union[str, DeviceClassName, DeviceName]" has no attribute "lower"  [union-attr]
homeassistant/components/shelly/utils.py:91: error: Item "DeviceName" of "Union[str, DeviceClassName, DeviceName]" has no attribute "lower"  [union-attr]
```